### PR TITLE
Refactor ServiceStartupPolicy to decrease execution time

### PR DESCRIPTION
**Pull request by marinakog**
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7016/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/6997

Split  to 3 different classes:
 
tests verifying Dynamic Cluster( ItServiceStartupPolicyDynamicCluster.java), 
tests verifying Config Cluster ( ItServiceStartupPolicyConfigCluster.java)
tests verifying generic standalone managed servers and domain -  ItServiceStartupPolicy.java),

### CODE REVIEW

**Review by anpanigr — APPROVED** [review]
lgtm

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java [hunk 1]
```diff
@@ -157,25 +108,25 @@ public static void initAll(@Namespaces(2) List<String> namespaces) {
     logger.info("Create secret for admin credentials");
     String adminSecretName = "weblogic-credentials";
     assertDoesNotThrow(() -> createDomainSecret(adminSecretName,"weblogic",
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java [hunk 2]
```diff
@@ -157,25 +108,25 @@ public static void initAll(@Namespaces(2) List<String> namespaces) {
...
-            "welcome1", domainNamespace),
-            String.format("createSecret failed for %s", adminSecretName));
+        "welcome1", domainNamespace),
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyConfigCluster.java [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+import static oracle.weblogic.kubernetes.utils.ServerStartPolicyUtils.scalingClusters;
+import static oracle.weblogic.kubernetes.utils.ServerStartPolicyUtils.setupSample;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests to verify that life cycle operation of config cluster does not impact the state of dynamic cluster.
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyConfigCluster.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyConfigCluster.java [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+
+  private static final String adminServerPodName = domainUid + "-admin-server";
+  private final String managedServerPrefix = domainUid + "-" + managedServerNamePrefix;
+  private static LoggingFacade logger = null;
+  private static String samplePath = "sample-testing-config-cluster";
+  private static String ingressHost = null; //only used for OKD
+
+  /**
+   * Install Operator.
+   * Create a domain resource definition.
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyConfigCluster.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyConfigCluster.java [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // Create the repo secret to pull the image
+    // this secret is used only for non-kind cluster
+    createOcirRepoSecret(domainNamespace);
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    String adminSecretName = "weblogic-credentials";
+    assertDoesNotThrow(() -> createDomainSecret(adminSecretName,"weblogic",
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyConfigCluster.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java [hunk 6]
```diff
@@ -0,0 +1,606 @@
...
+
+  private static final String adminServerPodName = domainUid + "-admin-server";
+  private final String managedServerPrefix = domainUid + "-" + managedServerNamePrefix;
+  private static LoggingFacade logger = null;
+  private static String ingressHost = null; //only used for OKD
+  private static String samplePath = "sample-testing-dynamic-cluster";
+
+  /**
+   * Install Operator.
+   * Create a domain resource definition.
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java [hunk 7]
```diff
@@ -0,0 +1,606 @@
...
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests to verify that life cycle operation of dynamic cluster does not impact the state of config cluster.
+ */
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@DisplayName("ServerStartPolicy attribute in different levels in a MII domain")
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java [hunk 8]
```diff
@@ -0,0 +1,606 @@
...
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // Create the repo secret to pull the image
+    // this secret is used only for non-kind cluster
+    createOcirRepoSecret(domainNamespace);
+
+    // create secret for admin credentials
+    logger.info("Create secret for admin credentials");
+    String adminSecretName = "weblogic-credentials";
+    assertDoesNotThrow(() -> createDomainSecret(adminSecretName,"weblogic",
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java [hunk 9]
```diff
@@ -0,0 +1,606 @@
...
+        "welcome1", domainNamespace),
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java [hunk 10]
```diff
@@ -0,0 +1,606 @@
...
+  private static String samplePath = "sample-testing-dynamic-cluster";
+
+  /**
+   * Install Operator.
+   * Create a domain resource definition.
+   * @param namespaces list of namespaces created by the IntegrationTestWatcher by the
+  JUnit engine parameter resolution mechanism
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(2) List<String> namespaces) {
```
[the remark this answers is not in the document]
Comment (marinakog) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicyDynamicCluster.java:94]: done

**Review by sankarpn — APPROVED** [review]
LGTM

**Pull request merged**